research-software-directory/RSD-as-a-service
# 392 cookie consent

# Matomo cookie consent

Closes #392
Closes #474 

Changes proposed in this pull request:
* Add support for cookie consent when MATOMO_ID provided as env variable
* Enable user to accept or decline anonymous data collection
* Enable user to change the value on cookies page
* App footer layout: 
   * contact email moved to left side
   * eScience logo is removed from the footer (it is modified and not identical with official logo)
* Unused invitations component: left/right margin removed

How to test:
* `docker-compose build frontend` to build frontend
* provide MATOMO_ID and MATOMO_URL values in .env file, see .env.example
* `docker-comopse up` to start
* on any page except /cookies you will see popup at the bottom
![image](https://user-images.githubusercontent.com/9204081/187432556-3c36f203-61d8-4162-8a9f-4e275251a796.png)
* If you, somehow manage to navigate to cookies page before using accept/decliine buttons, the page will open with off value by default. 
* The cookies page looks like this. You can accept/decline anonymous data collection.
![image](https://user-images.githubusercontent.com/9204081/187433106-e7eacf60-8bac-46ce-95b4-c94deba7f100.png)
* you can remove cookies and reload the page and the cookie consent dialog should appear again.

## Note
Beside testing the functioning of cookie consent components, please review used texts in the consent message and on the cookies page. It would be great if we can further improve the text used.

PR Checklist:
